elf.c:34:18: error: use of undeclared identifier 'LIBELF_ARCH'

after make menuconfig; make clean; make

the build process stops at 
cd lib/elftoolchain-0.7.1/libelf;
do gcc -Wall -I/Users/jayne/Development/ws-git/elks/include -I/Users/jayne/Development/ws-git/elks/elks/include -I/Users/jayne/Development/ws-git/elks/elks/tools/elf2elks/lib/elftoolchain-0.7.1/libelf -I/Users/jayne/Development/ws-git/elks/elks/tools/elf2elks/lib/elftoolchain-0.7.1/common -D__KERNEL__ -c -I. -I../common "$src"; done;

elf.c:34:18: error: use of undeclared identifier 'LIBELF_ARCH'
        .libelf_arch            = LIBELF_ARCH,
                                  ^
elf.c:35:22: error: use of undeclared identifier 'LIBELF_BYTEORDER'
        .libelf_byteorder       = LIBELF_BYTEORDER,
                                  ^
elf.c:36:19: error: use of undeclared identifier 'LIBELF_CLASS'
        .libelf_class           = LIBELF_CLASS,
                                  ^

--- a/elks/tools/elf2elks/Makefile
+++ b/elks/tools/elf2elks/Makefile
@@ -57,10 +57,10 @@ elf2elks.o: elf2elks.c $(LIBELF)
 # make.  To avoid roping in BSD make as yet another dependency, we directly
 # run the commands to generate and compile the libelf sources here.  This
 # still requires m4 as a dependency though.
-$(LIBELF): $(ELFTOOLCHAIN).tar.bz2 lib/elftoolchain.patch
+$(LIBELF): $(ELFTOOLCHAIN).tar.bz2 lib/elftoolchain.patch lib/elftoolchain2.patch
 	rm -rf $(ELFTOOLCHAIN)
 	cd lib && tar -xvjf elftoolchain-0.7.1.tar.bz2
-	cd $(ELFTOOLCHAIN) && patch -p1 <'$(abspath lib/elftoolchain.patch)'
+	cd $(ELFTOOLCHAIN) && patch -p1 <'$(abspath lib/elftoolchain.patch)' && patch -p1 <'$(abspath lib/elftoolchain2.patch)'
 ifeq "Linux" "$(shell uname -s)"
 	set -e; \
 	cd $(ELFTOOLCHAIN)/common; \

--- a/elks/tools/elf2elks/lib/elftoolchain2.patch
+++ b/elks/tools/elf2elks/lib/elftoolchain2.patch
@@ -1,0 +1,13 @@
+--- elftoolchain-0.7.1/libelf/_libelf_config.h	2024-02-06 14:09:41.000000000 -0700
++++ 1	2024-02-06 14:09:27.000000000 -0700
+@@ -36,6 +36,10 @@
+ #define	LIBELF_ARCH		EM_386
+ #define	LIBELF_BYTEORDER	ELFDATA2LSB
+ #define	LIBELF_CLASS		ELFCLASS32
++#elif	defined(__aarch64__)
++#define	LIBELF_ARCH		EM_AARCH64
++#define	LIBELF_BYTEORDER	ELFDATA2LSB
++#define	LIBELF_CLASS		ELFCLASS64
+ #endif
+ 
+ #endif	/* __DragonFly__ */

